compiler/rustc_codegen_rmc/src/codegen/typ.rs

```diff
@@ -459,16 +464,16 @@ impl<'tcx> GotocCtx<'tcx> {
     /// We handle this by treating it as an incomplete struct.
     fn codegen_foreign(&mut self, ty: Ty<'tcx>, defid: DefId) -> Type {
         debug!("codegen_foreign {:?} {:?}", ty, defid);
-        let name = self.ty_mangled_name(ty);
-        self.ensure(&aggr_name(&name), |ctx, _| {
-            Symbol::incomplete_struct(&name, Some(ctx.ty_pretty_name(ty)))
+        let name = self.ty_mangled_name(ty).intern();
```

Comment:
Should `ty_mangled_name` just return an interned string?

Reply:
done